dbms/src/Common/LooseBoundedMPMCQueue.h

```diff
@@ -235,12 +306,12 @@ class LooseBoundedMPMCQueue
         queue.pop_back();
         current_auxiliary_memory_usage -= element.memory_usage;
         assert(!queue.empty() || current_auxiliary_memory_usage == 0);
-        writer_head.notifyNext();
```

Comment:
This change will move the `notifyNext` out of the critical area, is it an expected change?

Reply:
`WaitingNode::notifyNext` is changed to `std::condition_variable::notify_one` which does not need a lock.